js.go

```diff
@@ -798,14 +798,28 @@ func AckExplicit() SubOpt {
 	})
```

Comment:
we need to make sure that AckNone, AckAll is not attempted to be set for pull consumers too.

Reply:
Are you thinking something like this?
```go
func AckNone() SubOpt {
	return subOptFn(func(opts *subOpts) error {
		if o.pull > 0 {
			return fmt.Errorf("cannot set AckNone for pull consumers")
		}
		opts.cfg.AckPolicy = AckNonePolicy
		return nil
	})
}
```

If so, then I think this would require users pass `Pull(n)` before they pass `AckNone()` so that `AckNone` can check `o.pull` that `Pull` set.

Reply:
we could add the check to be done after the options have been processed on subscribe like: https://github.com/nats-io/nats.go/blob/master/js.go#L485-L488